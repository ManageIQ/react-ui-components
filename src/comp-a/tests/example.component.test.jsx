import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { ExampleCmp } from '../index';

test('ExampleCmp should be defined', () => {
  expect(ExampleCmp).toBeDefined();
});

test('Example test', () => {
  const component = renderer.create(<ExampleCmp />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Exmaple method test', () => {
  const component = shallow(<ExampleCmp />);
  expect(component.instance().bla()).toBe('some text');
});
