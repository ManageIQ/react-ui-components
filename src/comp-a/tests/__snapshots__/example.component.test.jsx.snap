// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example test 1`] = `
<div
  className="some-class"
>
  <div
    className="Select has-value is-clearable is-searchable Select--single"
    style={undefined}
  >
    <div
      className="Select-control"
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <span
        className="Select-multi-value-wrapper"
        id="react-select-2--value"
      >
        <div
          className="Select-value"
          style={undefined}
          title={undefined}
        >
          <span
            aria-selected="true"
            className="Select-value-label"
            id="react-select-2--value-item"
            role="option"
          >
            Error
          </span>
        </div>
        <div
          className="Select-input"
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <input
            aria-activedescendant="react-select-2--value"
            aria-describedby={undefined}
            aria-expanded="false"
            aria-haspopup="false"
            aria-label={undefined}
            aria-labelledby={undefined}
            aria-owns=""
            className={undefined}
            id={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            role="combobox"
            style={
              Object {
                "boxSizing": "content-box",
                "width": "5px",
              }
            }
            tabIndex={undefined}
            value=""
          />
          <div
            style={
              Object {
                "height": 0,
                "left": 0,
                "overflow": "scroll",
                "position": "absolute",
                "top": 0,
                "visibility": "hidden",
                "whiteSpace": "pre",
              }
            }
          >
            
          </div>
        </div>
      </span>
      <span
        aria-label="Clear value"
        className="Select-clear-zone"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        title="Clear value"
      >
        <span
          className="Select-clear"
          dangerouslySetInnerHTML={
            Object {
              "__html": "&times;",
            }
          }
        />
      </span>
      <span
        className="Select-arrow-zone"
        onMouseDown={[Function]}
      >
        <span
          className="Select-arrow"
          onMouseDown={[Function]}
        />
      </span>
    </div>
  </div>
  <input
    className="form-control"
    id={undefined}
    onChange={[Function]}
    type={undefined}
    value="something"
  />
  <div
    className="alert alert-danger"
  >
    <span
      aria-hidden="true"
      className="pficon pficon-error-circle-o"
    />
    something
  </div>
</div>
`;
